plugins/modules/system/sysrc.py

```diff
@@ -0,0 +1,277 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2019 David Lundgren <dlundgren@syberisle.net>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+author:
+    - David Lundgren (@dlundgren)
+module: sysrc
+short_description: Manage FreeBSD using sysrc
+version_added: '2.10'
+description:
+    - Manages /etc/rc.conf for FreeBSD
+options:
+    name:
+        description:
+            - Name of variable in C(/etc/rc.conf) to manage.
+        type: str
+        required: true
+    value:
+        description:
+            - The value to set when I(state=present)
+            - The value to add when I(state=value_present)
+            - The value to remove when I(state=value_absent)
+        type: str
+    state:
+        description:
+            - Use I(present) to add the variable.
+            - Use I(absent) to remove the variable.
+            - Use I(value_present) to add the value to the existing variable.
+            - Use I(value_absent) to remove the value from the existing variable.
+        type: str
+        default: "present"
+        choices: [ absent, present, value_present, value_absent ]
+    path:
+        description:
+            - Path to file to use instead of C(/etc/rc.conf).
+        type: str
+        default: "/etc/rc.conf"
+    delim:
+        description:
+            - Delimiter to be used insetad of C( )
+            - Only used when I(state=value_present) or I(state=value_absent).
+        default: " "
+        type: str
+    jail:
+        description:
+            - Name or ID of the jail to operate on
+        required: false
+        type: str
+notes:
+  - The C(name) cannot use . (periods) as sysrc doesn't support OID style names
+'''
+
+EXAMPLES = r'''
+---
+# enable mysql in the /etc/rc.conf
+- name: Configure mysql pid file
+  sysrc:
+    name: mysql_pidfile
+    value: "/var/run/mysqld/mysqld.pid"
+
+# enable accf_http kld in the boot loader
+- name: enable accf_http kld
+  sysrc:
+    name: accf_http_load
+    state: present
+    value: "YES"
+    path: /boot/loader.conf
+
+# add gif0 to cloned_interfaces
+- name: add gif0 interface
+  sysrc:
+    name: cloned_interfaces
+    state: value_present
+    value: "gif0"
+
+# enable nginx on a jail
+- name: add gif0 interface
+  sysrc:
+    name: nginx_enable
+    value: "YES"
+    jail: www
+'''
+
+RETURN = r'''
+changed:
+  description: Return changed for sysrc actions as true or false.
+  returned: always
+  type: bool
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import re
+
+
+class sysrc(object):
```

Comment:
```suggestion
class Sysrc(object):
```
In Python, class names usually start with upper-case letters.